### PR TITLE
RSDK-7885: Send Event to Run Networking Tests on Release Candidate

### PR DESCRIPTION
**Pull request by danielbotros**
### Description
* **RSDK-7885:** Run Networking Tests on release Candidate
* See [this pr](https://github.com/viamrobotics/sdk-integration-tests/pull/35)
  * On RC build and upload success, sends an event to sdk-integration-test repo which triggers Networking Tests workflow for rc builds
### Testing
* Running this command will trigger Networking Tests in the `sdk-integration-tests` repo:
```shell
gh api \       
  -X POST \
  -H "Accept: application/vnd.github+json" \
  /repos/viamrobotics/sdk-integration-tests/dispatches \
  -f event_type=new_rc_branch
```

### CODE REVIEW

File: .github/workflows/releasecandidate.yml [hunk 1]
```diff
@@ -67,3 +67,17 @@ jobs:
...
+  run-sdk-networking-tests:
+    needs: [test, staticbuild, appimage]
+    if: ${{ success() }}
+    name: Trigger Networking Tests Workflow
+    runs-on: ubuntu-latest
+    steps:
+      - name: Trigger Workflow
+        run: |
+          curl -X POST \
+          -H "Authorization: token ${{secrets.GITHUB_TOKEN}}" \
```
Comment (danielbotros) [inline .github/workflows/releasecandidate.yml:80]: I'm not sure if this is the right token needed to be able to run this curl, it needs to have multiple repo write access to work. Theoretically with the token I could run this command to see if it triggers the workflow as well

**Review by benjirewis — COMMENTED** [review]
Nice!

File: .github/workflows/releasecandidate.yml [hunk 2]
```diff
@@ -67,3 +67,17 @@ jobs:
...
+          -H "Accept: application/vnd.github.everest-preview+json" \
```
Comment (abe-winter) [inline .github/workflows/releasecandidate.yml:81]: per [this](https://developer.github.com/changes/3/) I think everest-preview is no longer current

from [here](https://docs.github.com/en/rest/repos/repos?apiVersion=2022-11-28#create-a-repository-dispatch-event) I think it's okay to use vnd.github+json now

File: .github/workflows/releasecandidate.yml (hunk 1 above)
Comment (abe-winter) [reply .github/workflows/releasecandidate.yml:80]: another testing option -- make this its own file, give it a workflow_dispatch event, and run manually

(no harm in running extra sdk tests)

I have no preference though, do whatever works best for sdk team.

File: .github/workflows/releasecandidate.yml [hunk 3]
```diff
@@ -67,3 +67,17 @@ jobs:
           slack_webhook_url: ${{secrets.SLACK_WEBHOOK_URL}}
           channel: '#team-devops'
           name: 'Workflow Status'
+    
+  run-sdk-networking-tests:
+    needs: [test, staticbuild, appimage]
```
[the remark this answers is not in the document]
Comment (danielbotros) [reply .github/workflows/releasecandidate.yml:72]: I believe so as the last step of the `appimage` workflow is to output links to the newly uploaded RC builds, and this job won't run until `appimage` is successful

File: .github/workflows/releasecandidate.yml [hunk 4]
```diff
@@ -67,3 +67,18 @@ jobs:
...
+    runs-on: ubuntu-latest
+    steps:
+      - name: Trigger Workflow
+        run: |
+          curl -X POST \
+          -H "Authorization: token ${{secrets.GITHUB_TOKEN}}" \
+          -H "Accept: application/vnd.github+json" \
+          https://api.github.com/repos/viamrobotics/sdk-integration-tests/dispatches \
+          -d '{"event_type":"new_rc_branch"}'
+
```
Comment (benjirewis) [inline .github/workflows/releasecandidate.yml:84]: Something's up with your editor here, as there's still no terminating newline. Maybe try https://stackoverflow.com/questions/44704968/visual-studio-code-insert-newline-at-the-end-of-files?

Comment (danielbotros) [reply .github/workflows/releasecandidate.yml]: thanks for the tip! will never have this issue again 😼

**Review by benjirewis — APPROVED** [review]
LGTM as long as we're confident the `secrets.REPO_READ_TOKEN` will correctly authorize the curl 🧑‍🔧 .

**Comment by maximpertsov** [comment]
> LGTM as long as we're confident the `secrets.REPO_READ_TOKEN` will correctly authorize the curl 🧑‍🔧 .

If it doesn't, will this workflow be marked as a failure? And if so, does that mess up any release workflows downstream? @abe-winter

**Review by abe-winter — COMMENTED** [review]
would normally say 'merge + test in prod' but it would be bad to break the RC workflow

imo it's worth finding a way to test this that isn't 'run an RC' (RCs are kind of stateful because of the go sumdb)

recommendation: make this its own yaml file with `on: workflow_call` and `workflow_dispatch`, merge to main, run manually + see what happens; then invoke via workflow call from releasecandidate.yml file

(can talk through live if helpful)

**Review by maximpertsov — APPROVED** [review]
The PR looks good pending testing suggested here https://github.com/viamrobotics/rdk/pull/4083#pullrequestreview-2130819245

**Pull request merged**